controllers/machinedeployment_controller.go

```diff
@@ -95,6 +99,7 @@ func (r *MachineDeploymentReconciler) SetupWithManager(ctx context.Context, mgr
 
 func (r *MachineDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
 	log := ctrl.LoggerFrom(ctx)
+	r.reconciliationTime = metav1.Now()
```

Comment:
This is set throughout all concurrent reconciliation and should probably not be on the struct